src/NHibernate/Engine/Query/HQLQueryPlan.cs

```diff
@@ -174,6 +198,69 @@ public IEnumerable PerformIterate(QueryParameters queryParameters, IEventSource
 			return new JoinedEnumerable(results);
 		}
 
+		// Since v5.3
+		[Obsolete("This method has no more usages and will be removed in a future version")]
+		public async Task<IEnumerable> PerformIterateAsync(QueryParameters queryParameters, IEventSource session, CancellationToken cancellationToken)
+		{
+			cancellationToken.ThrowIfCancellationRequested();
+			if (Log.IsDebugEnabled())
+			{
+				Log.Debug("enumerable: {0}", _sourceQuery);
+				queryParameters.LogParameters(session.Factory);
+			}
+			if (Translators.Length == 0)
+			{
+				return CollectionHelper.EmptyEnumerable;
+			}
+			if (Translators.Length == 1)
+			{
+				return await (Translators[0].GetEnumerableAsync(queryParameters, session, cancellationToken)).ConfigureAwait(false);
+			}
+			var results = new IEnumerable[Translators.Length];
+			for (int i = 0; i < Translators.Length; i++)
+			{
+				var result = await (Translators[i].GetEnumerableAsync(queryParameters, session, cancellationToken)).ConfigureAwait(false);
+				results[i] = result;
+			}
+			return new JoinedEnumerable(results);
+		}
+
+		// Since v5.3
+		[Obsolete("This method has no more usages and will be removed in a future version")]
```

Comment:
It seems enough to have async methods in interface marked as obsolete. Generator seems smart enough to find obsolete async methods in interface and keep them generated. That would make less change noise.

Reply:
I overlooked that, now it is generated instead.